Remotion/Web/ClientScript/Scripts/Utilities.ts

```diff
@@ -377,44 +377,131 @@ class LayoutUtility
+  }
+
+  public static GetOuterWidth(element: HTMLElement)
+  {
+    return this.GetElementDimensionWithElementShow(element, "width", "border");
+  }
+
+  private static GetElementDimensionWithElementShow(element: Element, dimension: "width" | "height", extra?: "padding" | "border")
+  {
+    const htmlElement = element as HTMLElement;
+    const offsetValue = dimension === "width" ? htmlElement.offsetWidth : htmlElement.offsetHeight;
+    if (offsetValue !== 0)
+      return this.GetElementDimension(htmlElement, dimension, extra);
+
+    const previousPosition = htmlElement.style.position;
+    const previousVisibility = htmlElement.style.visibility;
+    const previousDisplay = htmlElement.style.display;
+
+    htmlElement.style.position = "absolute";
+    htmlElement.style.visibility = "hidden";
+    htmlElement.style.display = "block";
+
+    const result = this.GetElementDimension(htmlElement, dimension, extra);
+
+    htmlElement.style.position = previousPosition;
+    htmlElement.style.visibility = previousVisibility;
+    htmlElement.style.display = previousDisplay;
+
+    return result;
+  }
+
+  private static GetElementDimension(element: HTMLElement, dimension: "width" | "height", extra?: "padding" | "border")
   {
     const style = window.getComputedStyle(element);
 
-    // jQuery 1.6.4 uses the offsetWidth/offsetHeight properties to calculate element dimensions.
-    // The offsetXXX properties are rounded, while the style properties are not.
-    // We round here to make sure that we are compatible with the original behavior, but we do
-    // not use the offsetXXX properties for the calculation because they introduce unnecessary
-    // complexity for backwards compatibility that we don't need (IE).
-    const value = Math.round(parseFloat(style[dimension])) || 0;
-    
-    // For content-box: value is already correct
-    // For border-box: we need to remove padding and border from the value
-    if (style.boxSizing !== "border-box")
-      return value;
-    
     const orientations: CssOrientation[] = dimension === "width"
       ? ["Left", "Right"]
       : ["Top", "Bottom"];
 
+    const offsetValue = dimension === "width" ? element.offsetWidth : element.offsetHeight;
+    if (offsetValue > 0) 
+    {
+      if (extra === "border")
+        return offsetValue;
+
+      let adjustment = 0;
+      for (const orientation of orientations)
+      {
+        if (extra === undefined)
+        {
+          // Subtract padding
+          const paddingProperty = `padding${orientation}` as `padding${CssOrientation}`;
+          adjustment -= parseFloat(style[paddingProperty]) || 0;
+        }
+
+        // Subtract border
+        const borderProperty = `border${orientation}Width` as `border${CssOrientation}Width`;
+        adjustment -= parseFloat(style[borderProperty]) || 0;
+      }
+
+      return offsetValue + adjustment;
+    }
+
+    const value = parseFloat(style[dimension]) || 0;
```

Comment:
I would prefer to start an else block here. The IF isn't just a guard condition for an early exit but rather an entire branch of exeution logic so the two blocks should be connected.